### PR TITLE
chore: enable new jsx transform

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

This was added in react@17 and is more performant and removes the need to import `React` in every file containing JSX.

It also removes the need to set DISABLE_NEW_JSX_TRANSFORM which sometimes caused the tsconfig.json to flap a change.

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

### CODE REVIEW

File: src/containers/NFT/test/NFT.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import * as React from 'react'
```
Comment (mvadari) [inline src/containers/NFT/test/NFT.test.js:1]: Why is this needed?

File: src/containers/shared/EmptyMessageTableRow.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import * as React from 'react'
```
Comment (mvadari) [inline src/containers/shared/EmptyMessageTableRow.tsx:1]: ditto

File: src/containers/shared/components/Transaction/SimpleRow.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import * as React from 'react'
```
Comment (mvadari) [inline src/containers/shared/components/Transaction/SimpleRow.tsx:1]: ditto

File: src/containers/shared/components/Transaction/test/createWrapperFactory.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { mount, ReactWrapper } from 'enzyme'
-import React from 'react'
+import * as React from 'react'
```
Comment (mvadari) [inline src/containers/shared/components/Transaction/test/createWrapperFactory.tsx:2]: ditto

File: src/containers/shared/components/Transaction/types.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import * as React from 'react'
```
Comment (mvadari) [inline src/containers/shared/components/Transaction/types.ts:1]: ditto

File: src/containers/NFT/test/NFT.test.js (hunk 1 above)
Comment (ckniffen) [reply src/containers/NFT/test/NFT.test.js:1]: Due to the way the test is mocking `useState`.  I was able to update all the other places though.  Good catch there were 12 places.  I used a script that was supposed to rename all of those but I guess it missed a few.

File: src/containers/CustomNetworkHome/index.tsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { KeyboardEvent, MouseEvent, useState } from 'react'
+import { KeyboardEvent, MouseEvent as ReactMouseEvent, useState } from 'react'
```
Comment (mvadari) [inline src/containers/CustomNetworkHome/index.tsx:1]: Does importing it as `MouseEvent` not work?

Comment (ckniffen) [reply src/containers/CustomNetworkHome/index.tsx:1]: No because we also need the native MouseEvent type in the file.

**Review by mvadari — APPROVED** [review]

**Review by khancode — APPROVED** [review]
